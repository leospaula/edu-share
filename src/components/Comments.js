import React from 'react'

const Comments = props => (
  <div className="my-3">
    <h5>Comments:</h5>
    <ul className="list-group-flush px-0">
      {props.comments.map(comment => (
        <li className="list-group-item quartinary-background pt-3">
          <h6 className="card-subtitle text-muted">{comment.user.username ? comment.user.username : props.user.current.username}: </h6>
          <span>{comment.text}</span>
        </li>
      ))}
    </ul>
  </div>
)

export default Comments