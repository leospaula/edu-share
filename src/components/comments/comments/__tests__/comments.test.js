import React from 'react'
import ReactDOM from 'react-dom'
import Comments from '../Comments'
import renderer from 'react-test-renderer'

const comments = [
  {
    id: 4,
    text: 'My kids love this!',
    resource_id: 4,
    user_id: 3,
    created_at: '2020-10-21T16:54:41.856Z',
    updated_at: '2020-10-21T16:54:41.856Z',
    user: {
      username: 'Naninbah'
    }
  },
  {
    id: 5,
    text: 'This is a test comment!',
    resource_id: 4,
    user_id: 1,
    created_at: '2020-10-21T16:54:41.856Z',
    updated_at: '2020-10-21T16:54:41.856Z',
    user: {
      username: 'Charlie'
    }
  }
]

it ('renders without crashing', ()=> {
  const div = document.createElement('div')
  ReactDOM.render(<Comments comments={comments}/>, div)
})

it ('matches the snapshot given mocked comments data', ()=> {
  const tree = renderer.create(<Comments comments={comments}/>).toJSON()
  expect(tree).toMatchSnapshot()
})

it ('displays the correct number of comments', ()=> {

})

it ('displays the name of the user who made the comment', ()=> {

})

it ('displays the text of the comment', ()=> {

})