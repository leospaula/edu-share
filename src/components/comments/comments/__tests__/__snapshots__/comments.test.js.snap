// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot given mocked comments data 1`] = `
<div
  className="my-3 comments-wrapper"
>
  <h5>
    Comments:
  </h5>
  <ul
    className="list-group-flush px-0"
  >
    <li
      className="list-group-item quartinary-background pt-3"
    >
      <h6
        className="card-subtitle text-muted"
      >
        Naninbah
        : 
      </h6>
      <span>
        My kids love this!
      </span>
    </li>
    <li
      className="list-group-item quartinary-background pt-3"
    >
      <h6
        className="card-subtitle text-muted"
      >
        Charlie
        : 
      </h6>
      <span>
        This is a test comment!
      </span>
    </li>
  </ul>
</div>
`;
